### PR TITLE
Windows/Shared: Set system mouse cursor #99

### PR DESCRIPTION
**Pull request by EgorOrachyov**

### CODE REVIEW

File: shared/java/org/jetbrains/jwm/MouseCursor.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+public enum MouseCursor {
+    ARROW  (0),
+    CROSS  (1),
+    HAND   (2),
+    HELP   (3),
+    IBEAM  (4),
+    UPARROW(5);
+
+    @ApiStatus.Internal public final int _id;
```
Comment (tonsky) [inline shared/java/org/jetbrains/jwm/MouseCursor.java:13]: Let’s use `_values` to transform enum <-> int, just for consistency with Key.java

File: shared/java/org/jetbrains/jwm/MouseCursor.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package org.jetbrains.jwm;
+
+import org.jetbrains.annotations.*;
+
+public enum MouseCursor {
+    ARROW  (0),
+    CROSS  (1),
+    HAND   (2),
```
Comment (tonsky) [inline shared/java/org/jetbrains/jwm/MouseCursor.java:8]: macOS has three types of hands: open, closed and pointing. Let’s have more specific name here? E.g. HAND_POINTING, HAND_OPEN or HAND_CLOSED?

File: shared/java/org/jetbrains/jwm/MouseCursor.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package org.jetbrains.jwm;
+
+import org.jetbrains.annotations.*;
+
+public enum MouseCursor {
+    ARROW  (0),
+    CROSS  (1),
```
Comment (tonsky) [inline shared/java/org/jetbrains/jwm/MouseCursor.java:7]: How about CROSSHAIR instead of CROSS?

**Review by tonsky — CHANGES_REQUESTED** [review]
Looks good, few minor naming suggestions

File: shared/java/org/jetbrains/jwm/MouseCursor.java (hunk 1 above)
Comment (EgorOrachyov) [reply shared/java/org/jetbrains/jwm/MouseCursor.java:13]: ID used to transform java enum to c++. 
There is no need to transform c++ to java => no values needed

File: shared/java/org/jetbrains/jwm/MouseCursor.java (hunk 2 above)
Comment (EgorOrachyov) [reply shared/java/org/jetbrains/jwm/MouseCursor.java:8]: Ok, but on windows they all will be mapped to IDC_HAND

Comment (tonsky) [reply shared/java/org/jetbrains/jwm/MouseCursor.java]: Let's leave only one, the one that Windows has?

File: shared/java/org/jetbrains/jwm/MouseCursor.java (hunk 1 above)
Comment (tonsky) [reply shared/java/org/jetbrains/jwm/MouseCursor.java:17]: IBEAM.ordinal() will get you the integer

Comment (EgorOrachyov) [reply shared/java/org/jetbrains/jwm/MouseCursor.java]: 👌

File: shared/java/org/jetbrains/jwm/MouseCursor.java (hunk 2 above)
Comment (EgorOrachyov) [reply shared/java/org/jetbrains/jwm/MouseCursor.java]: Windows has IDC_HAND

Comment (EgorOrachyov) [reply shared/java/org/jetbrains/jwm/MouseCursor.java]: Or we can do something like that in win

image

**Pull request merged**